extensions/notebook/src/book/bookTreeView.ts

```diff
@@ -190,17 +190,37 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 
 	async showPreviewFile(urlToOpen?: string): Promise<void> {
 		if (this.currentBook) {
-			const bookRoot = this.currentBook.bookItems[0];
-			const sectionToOpen = bookRoot.findChildSection(urlToOpen);
-			const urlPath = sectionToOpen ? sectionToOpen.url : bookRoot.tableOfContents.sections[0].url;
-			const sectionToOpenMarkdown: string = path.join(this.currentBook.bookPath, Content, urlPath.concat('.md'));
-			// The Notebook editor expects a posix path for the resource (it will still resolve to the correct fsPath based on OS)
-			const sectionToOpenNotebook: string = path.posix.join(this.currentBook.bookPath, Content, urlPath.concat('.ipynb'));
-			if (await fs.pathExists(sectionToOpenMarkdown)) {
-				this.openMarkdown(sectionToOpenMarkdown);
+			let urlPath: string;
+			if (this.currentBook.isNotebook) {
+				urlPath = urlToOpen && this.currentBook.bookPath === urlToOpen ? this.currentBook.bookPath : undefined;
+			} else {
+				if (urlToOpen) {
+					const bookRoot = this.currentBook.bookItems[0];
+					const sectionToOpen = bookRoot.findChildSection(urlToOpen);
+					urlPath = sectionToOpen?.url;
+				} else {
+					urlPath = this.currentBook.bookItems[0].tableOfContents.sections[0].url;
+				}
 			}
-			else if (await fs.pathExists(sectionToOpenNotebook)) {
-				await this.openNotebook(sectionToOpenNotebook);
+			if (urlPath) {
+				if (this.currentBook.isNotebook) {
```

Comment:
You're right it could be a markdown/ipynb file within a Book/NotebookFolder. If it's within a notebook folder, url has the exact path of the file but if it's part of a book, it's the url from toc file which doesn't include the file type so the extra logic.